source/containers/Form/hooks/textReplacement.ts

```diff
@@ -190,38 +205,52 @@ export const replaceText = (
   user: User,
   period?: FormPeriod,
   partner?: PartnerInfo,
-  encryptionPin?: string
-) => {
+  encryptionPin?: string,
+  completionsClarificationMessage?: string
+): string => {
   // This way of doing it might be a bit overkill, but the idea is that this in principle
   // allows for nesting replacement rules and then applying them in order one after the other.
   let res = text ?? "";
   replacementRules.forEach(([template, descriptor]) => {
     res = res.replace(
       template,
-      computeText(descriptor, user, period, partner, encryptionPin)
+      computeText(
+        descriptor,
+        user,
+        period,
+        partner,
+        encryptionPin,
+        completionsClarificationMessage
+      )
     );
   });
   return res;
 };
 
 /**
  * Replaces the markdown as specified by a set of markdown rules and logic that we've defined.
- * @param steps the steps of the form
- * @param user the user object
- * @param period A period to use for dates
  */
 export const replaceMarkdownTextInSteps = (
   steps: Step[],
   user: User,
   period?: FormPeriod,
   partner?: PartnerInfo,
-  encryptionPin?: string
+  encryptionPin?: string,
+  completionsClarificationMessage?: string
 ): Step[] => {
+  console.log("replaceMarkdownTextInSteps:", completionsClarificationMessage);
   const newSteps = steps.map((step) => {
```

Comment:
**OPTIONAL**
You can save at least 70 rows of code by doing a small refactoring of `replaceMarkdownTextInSteps` function.
Notice the lambda function `replaceString`.

```js
export const replaceMarkdownTextInSteps = (
  steps: Step[],
  user: User,
  period?: FormPeriod,
  partner?: PartnerInfo,
  encryptionPin?: string,
  completionsClarificationMessage?: string
): Step[] => {
  console.log("replaceMarkdownTextInSteps:", completionsClarificationMessage);

  const replaceString = (text: string): string =>
    replaceText(
      text,
      user,
      period,
      partner,
      encryptionPin,
      completionsClarificationMessage
    );

  const newSteps = steps.map((step) => {
    if (step.questions) {
      step.questions = step.questions.map((qs) => {
        if (qs.text && qs.text !== "") {
          qs.text = replaceString(qs.text);
        }

        if (qs.label && qs.label !== "") {
          qs.label = replaceString(qs.label);
        }

        if (qs.title && qs.title !== "") {
          qs.title = replaceString(qs.title);
        }

        if (qs.heading && qs.heading !== "") {
          qs.heading = replaceString(qs.heading);
        }

        if (qs.items) {
          qs.items = qs.items.map((item) => ({
            ...item,
            title: replaceString(item.title),
          }));
        }

        if (qs.categories) {
          qs.categories = qs.categories.map((category) => ({
            ...category,
            description: replaceString(category.description),
          }));
        }

        if (qs.inputs) {
          qs.inputs = qs.inputs.map((input) => ({
            ...input,
            label: input.label && replaceString(input.label),
            title: input.title && replaceString(input.title),
          }));
        }

        if (qs.components) {
          qs.components = qs.components.map((input) => ({
            ...input,
            text: replaceString(input.text),
          }));
        }

        return qs;
      });
    }
    if (step.title) step.title = replaceString(step.title);
    if (step.description) step.description = replaceString(step.description);
    return step;
  });

  return newSteps;
};
```

Small change to do, and the outcome if quite nice